### PR TITLE
feat: add committee to centralized br

### PR DESCRIPTION
**Pull request by clbartoli**
The Centralized block relay should have a committee defined when deployed, so that not only onne address can post blocks

### CODE REVIEW

File: contracts/CentralizedBlockRelay.sol [hunk 1]
```diff
@@ -55,11 +57,32 @@ contract CentralizedBlockRelay is BlockRelayInterface {
...
+  {
+    for (uint256 i; i < committee.length; i++) {
+      if (committee[i] == _address) {
+      return true;
+
+      }
+        return false;
+      
+
+  }
```
Comment (girazoki) [inline contracts/CentralizedBlockRelay.sol:83]: FMT

File: contracts/CentralizedBlockRelay.sol [hunk 2]
```diff
@@ -55,11 +57,31 @@ contract CentralizedBlockRelay is BlockRelayInterface {
     _;
   }
 
-  constructor() public{
+  constructor(address[] memory _committee) public{
     // Only the contract deployer is able to push blocks
     witnet = msg.sender;
```
Comment (mariocao) [inline contracts/CentralizedBlockRelay.sol:62]: Maybe it is not required anymore! :)

File: contracts/CentralizedBlockRelay.sol [hunk 3]
```diff
@@ -55,11 +57,31 @@ contract CentralizedBlockRelay is BlockRelayInterface {
...
+    committee = _committee;
   }
 
+
+
+
```
Comment (mariocao) [inline contracts/CentralizedBlockRelay.sol:68]: fmt (too many new lines) :rofl:

File: contracts/CentralizedBlockRelay.sol [hunk 4]
```diff
@@ -55,11 +57,31 @@ contract CentralizedBlockRelay is BlockRelayInterface {
...
+  /// @return bytes to be signed by bridge nodes
+  function isInCommittee(address _address)
+    public
+  returns(bool)
+  {
+    for (uint256 i; i < committee.length; i++) {
+      if (committee[i] == _address) {
+
+      return true;
+
```
Comment (mariocao) [inline contracts/CentralizedBlockRelay.sol:79]:
```suggestion
```

File: test/cent_block_relay.js [hunk 5]
```diff
@@ -6,8 +6,9 @@ const testdata = require("./poi.json")
 contract("Centralized Block relay", accounts => {
   describe("Centralized Block relay test suite", () => {
     let blockRelayInstance
+    const committee = [accounts[0], accounts[1]]
     before(async () => {
-      blockRelayInstance = await CentralizedBlockRelay.new({
+      blockRelayInstance = await CentralizedBlockRelay.new(committee, {
```
Comment (mariocao) [inline test/cent_block_relay.js:11]: I would also test the other committee member :)

File: contracts/CentralizedBlockRelay.sol [hunk 6]
```diff
@@ -55,11 +57,31 @@ contract CentralizedBlockRelay is BlockRelayInterface {
...
+    public
+  returns(bool)
+  {
+    for (uint256 i; i < committee.length; i++) {
+      if (committee[i] == _address) {
+
+      return true;
+      
+      }
+        return false;
```
Comment (mariocao) [inline contracts/CentralizedBlockRelay.sol:81]: indentation?

File: contracts/CentralizedBlockRelay.sol (hunk 4 above)
Comment (mariocao) [reply contracts/CentralizedBlockRelay.sol:79]: Maybe this function should be inside the modifier `isAuthorized`

**Review by mariocao — CHANGES_REQUESTED** [review]
And you metioned to me, include the `isAuthorized` modifier to the `isUpgradeable` . :P

**Review by girazoki — APPROVED** [review]

File: contracts/CentralizedBlockRelay.sol [hunk 7]
```diff
@@ -37,9 +36,18 @@ contract CentralizedBlockRelay is BlockRelayInterface {
   // Event emitted when a new block is posted to the contract
   event NewBlock(address indexed _from, uint256 _id);
 
-  // Only the owner should be able to push blocks
-  modifier isOwner() {
-    require(msg.sender == witnet, "Sender not authorized"); // If it is incorrect here, it reverts.
+  event Prueba(uint256 _id);
```
Comment (mariocao) [inline contracts/CentralizedBlockRelay.sol:39]: You forgot to remove it :)

**Review by mariocao — APPROVED** [review]

**Pull request merged**